kiali.go

```diff
@@ -211,3 +222,25 @@ func determineContainerVersion(defaultVersion string) string {
 	}
 	return v
 }
+
+// checkVersionCompatibility check kiali version compatibility with mesh.
+// The user session is not affected no matter what this check returns, just warning logs.
+func checkVersionCompatibility(meshName string, meshVersion string) {
+	if meshName == "Unknown" {
+		log.Warningf("Unknown Istio Implementation")
+		status.Put(status.MeshVersion, meshVersion)
+		status.Put(status.MeshName, meshName)
+		status.AddWarningMessages("Unknown Istio Implementation")
+	} else {
+		if ok := status.CheckMeshVersion(meshName, meshVersion, version); ok {
+			log.Infof("Mesh Name: %v Mesh Version: %v", meshName, meshVersion)
+			status.Put(status.MeshVersion, meshVersion)
+			status.Put(status.MeshName, meshName)
+			status.Put(status.IsCompatible, "true")
+		} else {
+			log.Warningf("Kiali %v is not compatible with %v %v", version, meshName, meshVersion)
```

Comment:
This message is fine for OSSM and Maistra, but for upstream Istio it may be too agressive.  We know it's not recommended (because it's not tested) but we may not be sure if it's incompatible.   I would suggest something like:

`log.Warningf("Kiali %v may not be compatible with %v %v, and is not recommended. See kiali.io for version compatibility", version, meshName, meshVersion)`

Reply:
Yeah, let's warn them that it MAY NOT be compatible. I think the suggested change is good.

Reply:
Done.